src/host/utf8handling.c

```diff
@@ -0,0 +1,33 @@
+/**
+ * \file   utf8handking.c
+ * \brief  Handles conversions between Unicode (UTF-8) and system native encoding (wide chars on Windows)
+ * \author Copyright (c) 2017 Jérôme Leclercq and the Premake project
+ */
+
+#include "premake.h"
+#include "stdlib.h"
+
+#ifdef PLATFORM_WINDOWS
+const char* utf8_fromwide(lua_State* L, const wchar_t* wstr)
+{
+	int size_required = WideCharToMultiByte(CP_UTF8, 0, wstr, -1, NULL, 0, NULL, NULL);
+
+	char* unicode_str = (char*) malloc(size_required * sizeof(char));
+	WideCharToMultiByte(CP_UTF8, 0, wstr, -1, unicode_str, size_required, NULL, NULL);
+
+	lua_pushstring(L, unicode_str);
+	free(unicode_str);
+	
+	return unicode_str;
```

Comment:
you just free'd that pointer.... should really not return that, as that will be a "use after free" access violation.